### PR TITLE
Add Logging Functionality ( #111 feat: log to a file)

### PR DESCRIPTION
**Pull request by Bigguysahaj**
This pull request addresses [Issue #111](https://github.com/ChakshuGautam/cQube-ingestion/issues/111).

**Changes Made:**
- Added `logger.ts` file in `c-qube/src/utils` directory.
- Implemented logging functionality using the popular `winston` library, including TypeScript typings (`@types/winston`).
- Chose `winston` for logging due to its flexibility and extensive feature set compared to a simple `fs` solution.
- Included an example of how to use the logger in the code snippet provided.

**Usage Example:**
```
const { setupLogger } = require('./utils/logger');

const logger = setupLogger('app.log', 'info');

logger.info('This is an info message');
logger.warn('This is a warning message');
logger.error('This is an error message');
```

Additionally, while working on this task, I noticed the `debug.ts` file, which appears to provide similar functionality. Please let me know if this implementation meets the requirements outlined in the issue, or if there is any other specific approach I should consider.

Please review and provide feedback. Thank you!

### CODE REVIEW

**Comment by Bigguysahaj** [comment]
@ChakshuGautam can you go through this?

**Comment by Bigguysahaj** [comment]
can I please review these changes @techsavvyash ?

File: impl/c-qube/src/utils/logger.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const fs = require('fs');
+
+function setupLogger(logFileName: string, logLevel: string = 'info') {
+  const logDir = 'logs';
+
+  if (!fs.existsSync(logDir)) {
+    fs.mkdirSync(logDir);
+  }
+
+  const logger = winston.createLogger({
```
Comment (techsavvyash) [inline impl/c-qube/src/utils/logger.ts:11]: Can you make the functionality for adding log levels available as function calls on the logger object? Similar to how it happens in the native nest js logger. 
Something like:
```ts
Logger.log('....');
Logger.error('...');
Logger.warn('...');
Logger.verbose('...');
```

**Review by techsavvyash — CHANGES_REQUESTED** [review]

**Comment by techsavvyash** [comment]
Hey @Bigguysahaj,
Since you are unresponsive to the changes required for this PR to become mergeable, I am marking this PR as closed. Please feel free to reopen it once you have finished it.
Thanks for your contribution

**Pull request closed without merging**